### PR TITLE
Allele subsetting code was broken when PL was missing. Got bad NPE.

### PR DESCRIPTION
**Pull request by yfarjoun**
### Description

-- fixed code
-- added test
----

@davidbenjamin could you please look at this?

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] Edited the README / documentation (if applicable)
- [ ] All tests passing on Travis

#### Review
- [ ] Final thumbs-up from reviewer
- [ ] Rebase, squash and reword as applicable

For more detailed guidelines, see https://github.com/broadinstitute/picard/wiki/Guidelines-for-pull-requests

### CODE REVIEW

File: src/main/java/picard/util/AlleleSubsettingUtils.java [hunk 1]
```diff
@@ -176,10 +176,15 @@ public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
                 }
             }
 
-            final boolean useNewLikelihoods = newPLs != null;
-            final GenotypeBuilder gb = useNewLikelihoods ? new GenotypeBuilder(g).PL(newPLs).log10PError(newLog10GQ) : new GenotypeBuilder(g).noPL().noGQ();
-            GenotypeLikelihoods.GenotypeLikelihoodsAllelePair allelePair = GenotypeLikelihoods.getAllelePair(MathUtil.indexOfMin(newPLs));
-            gb.alleles(Stream.of(allelePair.alleleIndex1, allelePair.alleleIndex2).map(allelesToKeep::get).collect(Collectors.toList()));
+            final GenotypeBuilder gb;
+            if (newPLs == null) {
+                gb = new GenotypeBuilder(g).noPL().noGQ().alleles(Arrays.asList(Allele.NO_CALL, Allele.NO_CALL));
```
Comment (davidbenjamin) [inline src/main/java/picard/util/AlleleSubsettingUtils.java]: This list already defined as `GATKVariantContextUtils.NO_CALL_ALLELES`.  Although you might want to change the name to `DIPLOID_NO_CALL_ALLELES`.

File: src/test/java/picard/util/AlleleSubsettingUtilsTest.java [hunk 2]
```diff
@@ -19,6 +19,7 @@
     final List<Allele> AA = Arrays.asList(Aref, Aref);
     final List<Allele> AC = Arrays.asList(Aref, C);
     final List<Allele> CC = Arrays.asList(C, C);
+    final List<Allele> noCall = Arrays.asList(Allele.NO_CALL, Allele.NO_CALL);
```
Comment (davidbenjamin) [inline src/test/java/picard/util/AlleleSubsettingUtilsTest.java]: Likewise, this is `GATKVariantContextUtils.NO_CALL_ALLELES`.

File: src/main/java/picard/util/AlleleSubsettingUtils.java [hunk 3]
```diff
@@ -176,10 +176,15 @@ public static GenotypesContext subsetAlleles(final GenotypesContext originalGs,
...
+            } else {
+                gb = new GenotypeBuilder(g).PL(newPLs).log10PError(newLog10GQ);
+                GenotypeLikelihoods.GenotypeLikelihoodsAllelePair allelePair = GenotypeLikelihoods.getAllelePair(MathUtil.indexOfMin(newPLs));
```
Comment (davidbenjamin) [inline src/main/java/picard/util/AlleleSubsettingUtils.java]: You don't need to get the allele pair, then the alleles list.  Instead, use `GenotypeLikelihoods.getAlleles(MathUtil.indexOfMin(newPLs), 2)` (which gets the intermediate allele pair behind the scenes).  The `2` is the ploidy.

File: src/main/java/picard/util/AlleleSubsettingUtils.java (hunk 1 above)
Comment (yfarjoun) [reply src/main/java/picard/util/AlleleSubsettingUtils.java]: we're in picard...yes?

Comment (yfarjoun) [reply src/main/java/picard/util/AlleleSubsettingUtils.java]: though, it's a good idea to make one copy of that list in picard as well....note taken.

File: src/main/java/picard/util/AlleleSubsettingUtils.java (hunk 3 above)
Comment (yfarjoun) [reply src/main/java/picard/util/AlleleSubsettingUtils.java]: ah, thanks.

File: src/test/java/picard/util/AlleleSubsettingUtilsTest.java (hunk 2 above)
Comment (yfarjoun) [reply src/test/java/picard/util/AlleleSubsettingUtilsTest.java]: https://youtu.be/nIaN9Koa9oM not Kansas anymore... :-)

File: src/main/java/picard/util/AlleleSubsettingUtils.java (hunk 1 above)
Comment (davidbenjamin) [reply src/main/java/picard/util/AlleleSubsettingUtils.java]: I guess we are in Picard, but yeah, I think it belongs as a `public static` somewhere.

File: src/test/java/picard/util/AlleleSubsettingUtilsTest.java (hunk 2 above)
Comment (davidbenjamin) [reply src/test/java/picard/util/AlleleSubsettingUtilsTest.java]: Then bring Topeka to Oz.

**Comment by yfarjoun** [comment]
probably htsjdk, not picard...so in the meantime, it's private here, and once I get it inhtsjdk, I can change it here... OK?

**Comment by davidbenjamin** [comment]
OK.

**Comment by yfarjoun** [comment]
we good here? @davidbenjamin ?

**Comment by davidbenjamin** [comment]
Yes.

**Pull request merged**